overlord/snapstate/snapmgr.go

```diff
@@ -502,6 +502,80 @@ func (m *SnapManager) ensureForceDevmodeDropsDevmodeFromState() error {
 	return nil
 }
 
+// ensureSnapdSnapTransition will migrate systems to use the "snapd" snap
+func (m *SnapManager) ensureSnapdSnapTransition() error {
+	m.state.Lock()
+	defer m.state.Unlock()
+
+	// check if snapd snap is installed
+	var snapst SnapState
+	err := Get(m.state, "snapd", &snapst)
+	if err != nil && err != state.ErrNoState {
+		return err
+	}
+	// nothing to do
+	if snapst.IsInstalled() {
+		return nil
+	}
+
+	// check if we should try the transition
+	if err := canInstallSnapdSnap(m.state); err != nil {
+		// do nothing here, if the snapd snap cannot be installed
```

Comment:
Same, would it make sense to log it with debug level?

Reply:
Same as above.